tests/Resources/os.addontest.js

```diff
@@ -0,0 +1,236 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2011-Present by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Ti, process */
+/* eslint no-unused-expressions: "off" */
+'use strict';
+const should = require('./utilities/assertions'); // eslint-disable-line no-unused-vars
+let os;
+
+describe('os', function () {
+	it('should be required as core module', function () {
+		os = require('os');
+		os.should.be.an.Object;
+	});
+
+	it('.EOL', () => {
+		os.should.have.a.property('EOL').which.is.a.String;
+		// TODO: Validate \n or \r\n based on platform!
+	});
+
+	describe('#arch()', () => {
+		it('is a function', function () {
+			os.arch.should.be.a.Function;
+		});
+
+		it('returns a String', function () {
+			os.arch().should.be.a.String;
+		});
+
+		// node values: 'arm', 'arm64', 'ia32', 'mips', 'mipsel', 'ppc', 'ppc64', 's390', 's390x', 'x32', and 'x64'.
+		// iOS values: "arm64", "armv7", "x86_64", "i386", "Unknown"
+		// Android values: "armeabi", "armeabi-v7a", "arm64-v8a", "x86", "x86_64", "mips", "mips64", "unknown"
+		// Windows values: "x64", "ia64", "ARM", "x86", "unknown"
```

Comment:
Why do they all have `unknown` included?

Reply:
I think it's just a fallback in case some never expected exception/usage happens.

This does remind me, that I enumerated the values because I wanted to see if I could/should converge the equivalent arches together to match what node reports.

Reply:
That might be a good idea. At least for stuff that's the same, e.g. the sim archs:
- `x86`, `i386` => `x32`
- `x86_64` => `x64`
Not sure what's the difference between the individual `arm` archs, though.

Reply:
Ah, i see you already pushed a commit. If `i386`, `x86` -> `ia32`, what's node's `x32` then?

Reply:
I modified our return values from Ti.Platform.architecture to match the equivalents from Node (which are all lowercase). And yes, os.arch just returns process.arch. As for the difference between it returning the compiled target versus actual, I don't think that can ever differ in our case. I suppose it could if they're running a 32-bit build of an app on a 64-bit Android device? 

my best read of "x32" is an attempt to make x86/ia32 more sane when dealing with code trying to access 64-bit stuff: https://lwn.net/Articles/456731/